app/javascript/components/aggregate_status_card.jsx

```diff
@@ -49,7 +49,10 @@ const AggregateStatusCard = ({ providerId, providerType }) => {
 
   return (
     <div className="aggregate_status">
-      {data.refreshStatus
+      {(data.refreshStatus && data.refreshStatus.default_cred
+       && (data.refreshStatus.default_cred[0].value !== 'Valid'
+        || !data.refreshStatus.last_refresh.stale
+         || data.refreshStatus.last_refresh.status !== 'Success'))
```

Comment:
it feels like we should move this down in the refreshStatus itself.  I'm thinking something like `data.refreshStatus.overallStatus`, then here you can just look at that one property.

Reply:
(we can also do this in a follow-up if the current code works - I just don't like seeing too much logic in the UI)

Reply:
Agree, we can improve code, i can do a followup pr later.